File: phpPOO/POOAula07/Luta.php
<?php
require_once './Lutador.php';
class Luta {
    
   private $desafiado;
   private $desafiante;
   private $rounds;
   private $aprovada;
   
   public function marcarLuta($l1,$l2){
       if($l1->getCategoria()== $l2->getCategoria() && $l1 != $l2){
           $this->aprovada = true;
           $this->desafiado = $l1;
           $this->desafiante = $l2;
       }else{
           $this->aprovada = false;
           $this->desafiado = null;
           $this->desafiante = null;
           
       }
   }
   public function lutar(){
       if($this->aprovada){
           print "---DESAFIADO---<br>";
           $this->desafiado->apresentar();
           print "---DESAFIANTE---</br>";
           $this->desafiante->apresentar();
           $vencedor = rand(0,2);
           switch($vencedor){
               case 0:
                   print "<p>Empate</p>";
                   $this->desafiado->empatarLuta();
                   $this->desafiante->empatarLuta();
                   break;
               case 1:
                   print "<p>Vencedor: ". $this->desafiado->getNome()."</p>";
                   print "<p>Perdedor: ". $this->desafiante->getNome()."</p>";
                   $this->desafiado->ganharLuta();
                   $this->desafiante->perderLuta();
                   break;

               case 2:
                   print "<p>Vencedor ". $this->desafiante->getNome()."</p>";
                   print"<p>Perdedor ". $this->desafiado->getNome()."</p>";
                   $this->desafiado->perderLuta();
                   $this->desafiante->ganharLuta();
                   break;
               
           }
       }else{
           print "<p>Luta não pode acontecer!</p>";
       }
   }
   public function getDesafiado(): Lutador {
       return $this->desafiado;
   }

   public function getDesafiante(): Lutador {
       return $this->desafiante;
   }

   public function getRounds() {
       return $this->rounds;
   }

   public function getAprovada() {
       return $this->aprovada;
   }

   public function setDesafiado(Lutador $desafiado): void {
       $this->desafiado = $desafiado;
   }

   public function setDesafiante(Lutador $desafiante): void {
       $this->desafiante = $desafiante;
   }

   public function setRounds($rounds): void {
       $this->rounds = $rounds;
   }

   public function setAprovada($aprovada): void {
       $this->aprovada = $aprovada;
   }


}

File: javaPOO/POOAula07/src/aula07poo/Luta.java

package aula07poo;

import java.util.Random;

public class Luta {
   private Lutador desafiado;
   private Lutador desafiante;
   private int rounds;
   private boolean aprovada;
   
   public void marcarLuta(Lutador l1,Lutador l2){
       if(l1.getCategoria().equals(l2.getCategoria()) && l1 != l2){
           this.aprovada = true;
           this.desafiado = l1;
           this.desafiante = l2;
       }else{
           this.aprovada = false;
           this.desafiado = null;
           this.desafiante = null;
       }
   }
   public void lutar(){
       if(this.aprovada){
           System.out.println("");
           System.out.println("---DESAFIADO---");
           this.desafiado.apresentar();
           System.out.println("");
           System.out.println("---DESAFIANTE---");
           this.desafiante.apresentar();
           Random ale = new Random();
           int vencedor = ale.nextInt(3); // 0, 1 ou 2
           System.out.println("");
           System.out.println("Resultado da Luta:");
           System.out.println("");
           switch(vencedor){
               case 0:
                   System.out.println("Empatou");
                   this.desafiado.empatarLuta();
                   this.desafiante.empatarLuta();
                   break;
               case 1:
                   System.out.println("Vencedor: " + this.desafiado.getNome());
                   System.out.println("Perdedor: " + this.desafiante.getNome());
                   this.desafiado.ganharLuta();
                   this.desafiante.perderLuta();
                   break;
               case 2:
                   System.out.println("Vencedor " + this.desafiante.getNome());
                   System.out.println("Perdedor " + this.desafiado.getNome());
                   this.desafiante.ganharLuta();
                   this.desafiado.perderLuta(); 
                   break;
           }
           System.out.println("");
       }else{
           System.out.println("A luta nao pode acontecer");
       }
   }

    public Lutador getDesafiado() {
        return desafiado;
    }

    public void setDesafiado(Lutador desafiado) {
        this.desafiado = desafiado;
    }

    public Lutador getDesafiante() {
        return desafiante;
    }

    public void setDesafiante(Lutador desafiante) {
        this.desafiante = desafiante;
    }

    public int getRounds() {
        return rounds;
    }

    public void setRounds(int rounds) {
        this.rounds = rounds;
    }

    public boolean isAprovada() {
        return aprovada;
    }

    public void setAprovada(boolean aprovada) {
        this.aprovada = aprovada;
    }
   
}
